### PR TITLE
(bugfix) fix rheakv state machine bug

### LINKED ISSUE #136: RheaKV状态机apply时多节点不一致漏洞
RheaKV的rocksDB状态机，比如put，当rocksdb的put成功，则给client返回success，异常，则给client返回错误。

我认为这里有漏洞。
首先，状态机Apply时，log已经同步大多数节点，那么leader apply失败，不代表follower也apply失败吧？

非必然失败，这种应该抛出异常，等待下次apply重试才是。
必然失败（能确定leader和follower必定apply失败），这种可以响应client失败。

### PR DESCRIPTION
**Pull request by fengjiachun**
### Motivation:

If rheakv caught a critical error, should halt the state machine  to avoid they inconsistencies between different nodes.

### Modification:

As above describe.

### Result:

Fixes #136

### CODE REVIEW

**Comment by fengjiachun** [comment]
Need to add some unit test

**Comment by killme2008** [comment]
存在冲突

**Comment by fengjiachun** [comment]
和 #154 冲突了，我先解决下

File: jraft-rheakv/rheakv-core/src/main/java/com/alipay/sofa/jraft/rhea/client/DefaultRheaKVCliService.java [hunk 1]
```diff
@@ -84,8 +84,7 @@ public Status rangeSplit(final long regionId, final long newRegionId, final Stri
             if (response.isSuccess()) {
                 return Status.OK();
             }
-            return new Status(-1, "fail to range split on region %d, error: %s", regionId, String.valueOf(response
-                .getError()));
+            return new Status(-1, "Fail to range split on region %d, error: %s", regionId, String.valueOf(response));
```
Comment (killme2008) [inline jraft-rheakv/rheakv-core/src/main/java/com/alipay/sofa/jraft/rhea/client/DefaultRheaKVCliService.java]: 不需要调用 String.valueOf 吧？  格式化会 %s 自动调用 toString

File: jraft-rheakv/rheakv-core/src/main/java/com/alipay/sofa/jraft/rhea/storage/KVStoreStateMachine.java [hunk 2]
```diff
@@ -80,62 +80,73 @@ public KVStoreStateMachine(Region region, StoreEngine storeEngine) {
...
-            kvStates.add(KVState.of(kvOp, done));
-            ++stCount;
-            it.next();
-        }
-        if (!kvStates.isEmpty()) {
-            final KVState first = kvStates.getFirstElement();
-            assert first != null;
-            batchApplyAndRecycle(first.getOpByte(), kvStates);
+        } catch (final Throwable t) {
+            it.setErrorAndRollback(index - applied, new Status(RaftError.EIO, "StateMachine meet critical error: %s.",
```
Comment (killme2008) [inline jraft-rheakv/rheakv-core/src/main/java/com/alipay/sofa/jraft/rhea/storage/KVStoreStateMachine.java]: RaftError 选择 ESTATEMACHINE 更合适，并且这里建议加下日志，打印出堆栈

**Pull request merged**